From: Matt Madison <matt@madison.systems>
Date: Sat, 20 Jan 2017 08:00:00 -0700
Subject: [PATCH] gstreamer: disable gtk-doc builds

The gtk-doc patch from OE-Core is for the makefiles that
come with much more recent versions of the gstreamer plugin
build system than the version on which the NVIDIA plugin is
based, and it would be difficult to retrofit, so for now
just disable gtk-doc builds.

Upstream-Status: Inappropriate
Signed-off-by: Matt Madison <matt@madison.systems>

---
 common/gtk-doc.mak | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: gst-egl/common/gtk-doc.mak
===================================================================
--- gst-egl.orig/common/gtk-doc.mak
+++ gst-egl/common/gtk-doc.mak
@@ -46,108 +46,8 @@ REPORT_FILES = \
 
 CLEANFILES = $(SCANOBJ_FILES) $(REPORT_FILES) $(DOC_STAMPS) doc-registry.xml
 
-if ENABLE_GTK_DOC
-all-local: html-build.stamp
-
-#### setup ####
-
-setup-build.stamp: $(content_files)
-	-@if test "$(abs_srcdir)" != "$(abs_builddir)" ; then \
-	    echo '  DOC   Preparing build'; \
-	    files=`echo $(DOC_MAIN_SGML_FILE) $(DOC_OVERRIDES) $(DOC_MODULE)-sections.txt $(DOC_MODULE).types $(content_files)`; \
-	    if test "x$$files" != "x" ; then \
-	        for file in $$files ; do \
-	            test -f $(abs_srcdir)/$$file && \
-	                cp -pu $(abs_srcdir)/$$file $(abs_builddir)/ || true; \
-	        done; \
-	    fi; \
-	fi
-	@touch setup-build.stamp
-
-#### scan ####
-
-# in the case of non-srcdir builds, the built gst directory gets added
-# to gtk-doc scanning; but only then, to avoid duplicates
-scan-build.stamp: $(HFILE_GLOB) $(CFILE_GLOB)
-	@echo '  DOC   Scanning header files'
-	@_source_dir='' ;						\
-	for i in $(DOC_SOURCE_DIR) ; do					\
-	    _source_dir="$${_source_dir} --source-dir=$$i" ;	        \
-	done ;							        \
-	gtkdoc-scan							\
-		$(SCAN_OPTIONS) $(EXTRA_HFILES)				\
-		--module=$(DOC_MODULE)					\
-		$${_source_dir}         				\
-		--ignore-headers="$(IGNORE_HFILES)"
-	@if grep -l '^..*$$' $(DOC_MODULE).types > /dev/null; then	\
-	    echo "  DOC   Introspecting gobjects"; \
-	    GST_PLUGIN_SYSTEM_PATH_1_0=`cd $(top_builddir) && pwd`		\
-	    GST_PLUGIN_PATH_1_0=						\
-	    GST_REGISTRY_1_0=doc-registry.xml				\
-	    $(GTKDOC_EXTRA_ENVIRONMENT)					\
-	    CC="$(GTKDOC_CC)" LD="$(GTKDOC_LD)"				\
-	    CFLAGS="$(GTKDOC_CFLAGS) $(CFLAGS)"				\
-	    LDFLAGS="$(GTKDOC_LIBS) $(LDFLAGS)"				\
-	    gtkdoc-scangobj --type-init-func="gst_init(NULL,NULL)"	\
-	        --module=$(DOC_MODULE) ;				\
-	else								\
-	    for i in $(SCANOBJ_FILES) ; do				\
-	       test -f $$i || touch $$i ;				\
-	    done							\
-	fi
-	@touch scan-build.stamp
-
-$(DOC_MODULE)-decl.txt $(SCANOBJ_FILES) $(DOC_MODULE)-sections.txt $(DOC_MODULE)-overrides.txt: scan-build.stamp
-	@true
-
-#### xml ####
-
-sgml-build.stamp: setup-build.stamp $(DOC_MODULE)-decl.txt $(SCANOBJ_FILES) $(DOC_MODULE)-sections.txt $(expand_content_files)
-	@echo '  DOC   Building XML'
-	@gtkdoc-mkdb --module=$(DOC_MODULE) --source-dir=$(DOC_SOURCE_DIR)  --expand-content-files="$(expand_content_files)" --main-sgml-file=$(DOC_MAIN_SGML_FILE) --output-format=xml $(MKDB_OPTIONS)
-	@cp ../version.entities xml
-	@touch sgml-build.stamp
-
-sgml.stamp: sgml-build.stamp
-	@true
-
-#### html ####
-
-html-build.stamp: sgml.stamp $(DOC_MAIN_SGML_FILE) $(content_files)
-	@echo '  DOC   Building HTML'
-	@rm -rf html
-	@mkdir html
-	@cp -pr xml html
-	@cp ../version.entities ./
-	@mkhtml_options=""; \
-	gtkdoc-mkhtml 2>&1 --help | grep  >/dev/null "\-\-verbose"; \
-	if test "$(?)" = "0"; then \
-	  if test "x$(V)" = "x1"; then \
-	    mkhtml_options="$$mkhtml_options --verbose"; \
-	  fi; \
-	fi; \
-	@gtkdoc-mkhtml 2>&1 --help | grep  >/dev/null "\-\-path"; \
-	if test "$(?)" = "0"; then \
-	  mkhtml_options=--path="$(abs_srcdir)"; \
-	fi; \
-	cd html && gtkdoc-mkhtml $$mkhtml_options $(MKHTML_OPTIONS) $(DOC_MODULE)-@GST_API_VERSION@ ../$(DOC_MAIN_SGML_FILE)
-	@rm -rf html/xml
-	@rm -f version.entities
-	@test "x$(HTML_IMAGES)" = "x" ||  ( cd $(srcdir) && cp $(HTML_IMAGES) $(abs_builddir)/html )
-	@echo '  DOC   Fixing cross-references'
-	@gtkdoc-fixxref --module=$(DOC_MODULE) --module-dir=html --html-dir=$(HTML_DIR) $(FIXXREF_OPTIONS)
-	@touch html-build.stamp
-
-clean-local-gtkdoc:
-	@rm -rf xml tmpl html
-# clean files copied for nonsrcdir templates build
-	@if test x"$(srcdir)" != x. ; then \
-	        rm -rf $(DOC_MODULE).types; \
-	fi
-else
 all-local:
 clean-local-gtkdoc:
-endif
 
 clean-local: clean-local-gtkdoc
 	@rm -f *~ *.bak
